packages/firebase_crashlytics/firebase_crashlytics/test/firebase_crashlytics_test.dart

```diff
@@ -237,16 +236,56 @@ void main() {
 
       test('without class', () async {
         final List<String> lines = <String>[
-          'package:firebase_crashlytics/test/main.dart 12  main'
+          '#0      main (package:firebase_crashlytics/test/main.dart:12)'
         ];
-        final List<Map<String, String>> elements = getStackTraceElements(lines);
+        final StackTrace trace = StackTrace.fromString(lines.join('\n'));
+        final List<Map<String, String>> elements = getStackTraceElements(trace);
         expect(elements.length, 1);
         expect(elements.first, <String, String>{
           'method': 'main',
           'file': 'package:firebase_crashlytics/test/main.dart',
           'line': '12',
         });
       });
+
+      test('with android obfuscated stack trace', () async {
+        final List<String> lines = <String>[
+          'Warning: This VM has been configured to produce stack traces that violate the Dart standard.',
+          '*** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***',
+          'pid: 1357, tid: 1415, name 1.ui',
+          'isolate_dso_base: 75f178181000, vm_dso_base: 75f178181000',
+          'isolate_instructions: 75f17818f000, vm_instructions: 75f178183000',
+          '    #00 abs 000075f17833027b virt 00000000001af27b _kDartIsolateSnapshotInstructions+0x1a127b',
```

Comment:
Would it be possible to make this an integration test instead?
I'm skeptical about the format staying the same, which could break the regexp.

If it's too much work, just leave a todo with me assigned.

Reply:
Yes, format can change. But tests are for checking a PR breaks the API design. If someone creates a PR and the tests fail, it should be the PR's problem. The PR must be handle that error or change the test.

If obfuscated stack trace format test included in this package, tests will fail in the future. PRs that unrelated to trace format will fail. They will probably comment out this test and add a todo about handling the new obfuscated format.

Instead we can watch `native_stack_traces` package or [`StackTrace::ToCString` function](https://github.com/dart-lang/sdk/blob/master/runtime/vm/object.cc#L24930). If there is a breaking change, a GitHub issue should be opened. This is better than adding the test.

Also, we need to change the CI workflow. We need to build 2 variants: not obfuscated and obfuscated. And test on both android and ios. Yes, this is too much work.

So what do you think?